### PR TITLE
Label_Microservice worker should use the new Model classes

### PR DESCRIPTION
**Pull request by jlewi**
* Most of the inference logic in the Worker class can be deleted and
we can instead use the newly created Model classes.

* Use an in cluster deployment service
* Log the model and thresholds.

* cli.py is a simple CLI to send pubsub requests
  * It is ended for development and debugging.

* Create a utility to parse issue specs of the form {org}/{repo}#number

* We need to initialize IssueLabelPredictor() inside the pubsub thread;
otherwise we have threading issues when trying to access the model.

* The universal model needs to apply appropriate thresholds to its predictions.

* Worker no longer needs to apply thresholding of predictions because that is
  now handled by the individual model classes.

* Add a method to worker apply_repo_config that handles repo specific behavior
specified in the YAML file in the repo

  * Apply aliases to the labels
  * If the config specifies which labels to predict then filter out all other
    labels.

* When commenting on issues about the predictions, the bot should reate a markdown table with the probabilities for the labels in the comment.

* Remove test_worker.py; the test is no longer relevant after the refactor.
  There is a TODO in worker.py to create an appropriate unittest for the new
  code.

* Change how the GitHub App Private Keys are handled.

  * The private key should be mounted as a secret and an environment variable
    should specify the path of the PEM key.

  * Stop storing the PEM key as an environment variable and then in python code writing it to a file.

  * Change the name of the environment variables to add the prefix "GITHUB_"
to make it more explanatory.

  * Move some of the helper methods for creating GitHubApp out of github_util
    and into the GitHubApp class

Related to: #70 combine universal and repo specific models.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/code-intelligence/85)

### CODE REVIEW

File: py/code_intelligence/util.py [hunk 1]
```diff
@@ -8,4 +11,18 @@ def write_items_to_json(output_file, results):
...
+  Args:
+    isue: An issue in the form {owner}/{repo}#{number}
+
+  Returns:
+    owner, repo, number
+  """
+  m = ISSUE_RE.match(issue)
+  if not m:
+    return None, None, None
+  return m.group(1), m.group(2), int(m.group(3))
```
Comment (hamelsmu) [inline py/code_intelligence/util.py]: Maybe return a named tuple or a Dict?

**Comment by hamelsmu** [comment]
/lgtm

Going ahead and stamping with LGTM b/c only had a minor question which is optional

File: py/code_intelligence/util.py (hunk 1 above)
Comment (jlewi) [reply py/code_intelligence/util.py]: Sorry looks like this got merged before I saw this comment.

**Pull request merged**